src/main/java/com/adaptavist/tm4j/jenkins/extensions/Instance.java

```diff
@@ -3,25 +3,33 @@
 import com.mashape.unirest.http.HttpResponse;
 import com.mashape.unirest.http.JsonNode;
 import com.mashape.unirest.http.exceptions.UnirestException;
+import com.mashape.unirest.request.body.MultipartBody;
 import java.io.File;
 
-public interface Instance {
+public abstract class Instance {
 
-    Boolean cloud();
+    public abstract Boolean cloud();
 
-    String name();
+    public abstract String name();
 
-    Boolean isValidCredentials();
+    public abstract Boolean isValidCredentials();
 
-    HttpResponse<JsonNode> publishCucumberFormatBuildResult(final String projectKey, final Boolean autoCreateTestCases, final File zip,
+    public abstract HttpResponse<JsonNode> publishCucumberFormatBuildResult(final String projectKey, final Boolean autoCreateTestCases, final File zip,
                                                             final ExpandedCustomTestCycle expandedCustomTestCycle) throws UnirestException;
 
-    HttpResponse<JsonNode> publishCustomFormatBuildResult(final String projectKey, final Boolean autoCreateTestCases, final File zip,
+    public abstract HttpResponse<JsonNode> publishCustomFormatBuildResult(final String projectKey, final Boolean autoCreateTestCases, final File zip,
                                                           final ExpandedCustomTestCycle expandedCustomTestCycle) throws UnirestException;
 
-    HttpResponse<JsonNode> publishJUnitFormatBuildResult(final String projectKey, final Boolean autoCreateTestCases, final File zip,
+    public abstract HttpResponse<JsonNode> publishJUnitFormatBuildResult(final String projectKey, final Boolean autoCreateTestCases, final File zip,
                                                          final ExpandedCustomTestCycle expandedCustomTestCycle) throws UnirestException;
 
-    HttpResponse<String> downloadFeatureFile(final String projectKey) throws UnirestException;
+    public abstract HttpResponse<String> downloadFeatureFile(final String projectKey) throws UnirestException;
 
+    protected HttpResponse<JsonNode> getBodyAsJsonOrThrowExceptionWithBody(final MultipartBody body) throws UnirestException {
+        try {
+            return body.asJson();
+        } catch (Exception e) {
+            throw new RuntimeException(body.asString().getBody());
```

Comment:
I think would be good to add the Exception in `RuntimeException`

Reply:
Yeah, I thought about that but I think it would add noise to the error message already returned from the API.

However, the `Exception` is too generic. I changed it to `Unirest` exceptions only, so we can be more precise about what errors we're catching.